src/components/NewGoalCriteria/NewGoalCriteria.tsx

```diff
@@ -88,40 +129,202 @@ const GoalCriteria: React.FC<Omit<GoalCriteriaProps, 'id'>> = ({ title, shortId,
+                label: tr('Edit'),
+                icon: <IconEdit1Outline size="xs" />,
+                handler: () => setMode('edit'),
+            });
+        }
+
+        if (!criteriaAsGoal(props) && onConvert) {
+            actions.push({
+                label: tr('Create as goal'),
+                icon: <IconTargetOutline size="xs" />,
+                handler: () => onConvert(props),
+            });
+        }
+
+        if (onRemove) {
+            actions.push({
+                label: tr('Delete'),
+                icon: <IconBinOutline size="xs" color={danger0} />,
+                color: danger0,
+                handler: () => onRemove(props),
+            });
+        }
+
+        return actions;
+    }, [onConvert, onRemove, onUpdate, props]);
+
+    const handleCriteriaCheck = useCallback(() => {
+        if (onCheck) {
+            return () => {
+                onCheck({ ...props, isDone: !props.isDone });
+            };
+        }
+
+        return undefined;
+    }, [props, onCheck]);
+
+    const [defaultMode, values]: ['simple' | 'goal', React.ComponentProps<typeof GoalCriteriaSuggest>['values']] =
+        useMemo(() => {
+            if (criteriaAsGoal(props)) {
+                return [
+                    'goal',
+                    {
+                        id: props.id,
+                        mode: 'goal',
+                        title: props.title,
+                        weight: props.weight ? `${props.weight}` : '',
+                        selected: {
+                            id: props.goal.goalId,
+                            title: props.title,
+                            stateColor: props.goal.stateColor || 0,
+                        },
+                    },
+                ];
+            }
+
+            return [
+                'simple',
+                {
+                    id: props.id,
+                    mode: 'simple',
+                    title: props.title,
+                    weight: props.weight ? `${props.weight}` : '',
+                },
+            ];
+        }, [props]);
+
+    return (
+        <TableRow className={classes.NewGoalCriteriaTableRow}>
+            {criteriaAsGoal(props) ? (
+                <GoalCriteria title={props.title} weight={props.weight} isDone={props.isDone} goal={props.goal} />
+            ) : (
+                <SimpleCriteria
+                    title={props.title}
+                    isDone={props.isDone}
+                    weight={props.weight}
+                    onCheck={handleCriteriaCheck()}
+                />
+            )}
+            {nullable(availableActions, (actions) => (
+                <TableCell width="3rem">
+                    <CriteriaActions actions={actions} />
+                </TableCell>
+            ))}
+            {nullable(mode === 'edit', () => (
+                <GoalFormPopupTrigger
+                    defaultVisible
+                    renderTrigger={({ ref }) => <div ref={ref} />}
```

Comment:
empty div? 
renderTrigger --> to optional params?

Reply:
Then form popup has not target for positioning